resource-aggregate/cqrs/eventstore/mongodb/save.go

```diff
@@ -129,7 +129,7 @@ func checkBeforeSave(events ...eventstore.Event) error {
 			if event.Version() != version+uint64(idx) {
 				return fmt.Errorf("invalid continues ascending events[%v].Version(%v))", idx, event.Version())
 			}
-			if event.IsSnapshot() {
+			if event.IsSnapshot() && idx < len(events)-1 {
```

Comment:
please describe what is why this new condition. It makes the code much less understandable now as there is some additional hacky condition for a reason we will forget.